lightwood/encoder/time_series/rnn.py

```diff
@@ -169,35 +174,35 @@ def prepare(self, priming_data, dependency_data=None, ts_analysis=None,
             priming_data = torch.stack([d for d in priming_data]).unsqueeze(-1).to(self.device)
 
         # merge all normalized data into a training batch
-        if dependency_data is not None:
-            normalized_tensors = []
-            for dep_name, dep_data in dependency_data.items():
-                if dep_name in self.grouped_by:
-                    continue
-                if dep_data['original_type'] in (dtype.integer, dtype.float):
-                    dep_data['group_info'] = {group: dependency_data[group]['data'] for group in self.grouped_by}
-                    data = torch.zeros((len(priming_data), lengths_data.max().int().item(), 1))
-                    all_idxs = set(range(len(data)))
-                    for group_name, normalizer in self.dep_norms[dep_name].items():
-                        if group_name != '__default':
-                            idxs, subset = get_group_matches(dep_data, normalizer.combination)
-                            normalized = normalizer.encode(subset).unsqueeze(-1)
-                            data[idxs, :, :] = normalized
-                            all_idxs -= set(idxs)
-                    if len(all_idxs) > 0 and '__default' in self.dep_norms[dep_name].keys():
-                        default_norm = self.dep_norms[dep_name]['__default']
-                        subset = [dep_data['data'][idx] for idx in list(all_idxs)]
-                        data[list(all_idxs), :, :] = torch.Tensor(default_norm.encode(subset)).unsqueeze(-1)
+        normalized_tensors = []
```

Comment:
Looked for a while, I don't get this bit of logic, maybe double-check it :?

Reply:
Logic is untouched, I simply un-indented it as the default value for `depedency_data` is now also a Dict. I will add the type hints, though.